src/main/java/efs/task/syntax/GuessNumberGame.java

```diff
@@ -13,10 +18,75 @@ public static void main(String[] args) {
     }
 
     public GuessNumberGame(String argument) {
-        //TODO: Implement the constructor
+        try {
+            this.M = Integer.parseInt(argument);
+            if(M <1 || M > UsefulConstants.MAX_UPPER_BOUND){
+                System.out.println("\'"+argument+"\'" + " wyszło poza zakres : <1,"+UsefulConstants.MAX_UPPER_BOUND+"> - NIEPOPRAWNY ARGUMENT");
+                throw new IllegalArgumentException();
+            }
+        }
+        catch (NumberFormatException e){
+            System.out.println("\'" + argument + "\'" + ", który podałeś: " + UsefulConstants.WRONG_ARGUMENT + " nie jest liczbą - NIEPOPRAWNY ARGUMENT");
+            throw new IllegalArgumentException(e);
+        }
+
+
     }
 
     public void play() {
-        //TODO: Implement the method that executes the game session
+        Random val = new Random();
+        Scanner in = new Scanner(System.in);
+        this.liczba = val.nextInt(1+M);
+        this.L = (int) Math.abs(Math.log(M)/Math.log(2)+1);
+        int proby=0;
+        do {
+            System.out.println("Zakres gry: <1,"+this.M+">.");
+            char[] temp = new char[L];
+            for (int i = 0; i < proby + 1; ++i) {
+                temp[i] += '*';
+            }
+            for(int i=proby +1 ;i<L;++i){
+                temp[i]+='.';
+            }
+            System.out.print("[");
+            for (int i=0;i<temp.length;++i){
+                System.out.print(temp[i]);
+            }
+            System.out.println("]");
+            System.out.println(UsefulConstants.GIVE_ME+" liczbę do zgadywania: ");
+            String liczba_uzytkownik = in.nextLine();
+            int zgadywanie;
+            try {
```

Comment:
_clean code_ za duży blok _try/catch_, w takim bloku powinno się znaleźć tylko to, co rzuca wyjątkiem